modules/core/src/main/java/org/apache/ignite/internal/processors/tracing/configuration/GridTracingConfiguration.java

```diff
@@ -0,0 +1,303 @@
+        TracingConfigurationParameters lbBasedTracingConfiguration =
+            scopeSpecificTracingConfiguration.get(coordinates.label());
+
+        // If scope + label specific was found — use it.
+        if (lbBasedTracingConfiguration != null)
+            return lbBasedTracingConfiguration;
+
+        // Retrieving scope specific tracing configuration.
+        TracingConfigurationParameters rawScopedTracingConfiguration = scopeSpecificTracingConfiguration.get(null);
+
+        // If scope specific was found — use it.
+        if (rawScopedTracingConfiguration != null)
+            return rawScopedTracingConfiguration;
+
+        // If neither scope + label specific nor just scope specific configuration was found —
+        // use scope specific default one.
+        return TracingConfiguration.super.retrieveConfiguration(coordinates);
+    }
+
+    /** {@inheritDoc} */
+    @SuppressWarnings("AssignmentOrReturnOfFieldWithMutableType") @Override
+    public @NotNull Map<TracingConfigurationCoordinates, TracingConfigurationParameters> retrieveConfigurations() {
+        DistributedMetaStorage metaStore;
+
+        try {
+            metaStore = ctx.distributedMetastorage();
+        }
+        catch (Exception e) {
+            log.warning("Failed to retrieve tracing configuration — meta storage is not available.");
+
+            return DEFAULT_CONFIGURATION_MAP;
+        }
+
+        if (metaStore == null) {
+            log.warning("Failed to retrieve tracing configuration — meta storage is not available.");
+
+            return DEFAULT_CONFIGURATION_MAP;
+        }
+
+        Map<TracingConfigurationCoordinates, TracingConfigurationParameters> res = new HashMap<>();
+
+        for (Scope scope : Scope.values()) {
+            String scopeSpecificKey = TRACING_CONFIGURATION_DISTRIBUTED_METASTORE_KEY_PREFIX + scope.name();
+
+            try {
+                for (Map.Entry<String, TracingConfigurationParameters> entry :
+                    ((Map<String, TracingConfigurationParameters>)metaStore.read(scopeSpecificKey)).entrySet()) {
+                    res.put(
+                        new TracingConfigurationCoordinates.Builder(scope).withLabel(entry.getKey()).build(),
+                        entry.getValue());
+                }
+            }
+            catch (IgniteCheckedException e) {
+                LT.warn(log, "Failed to retrieve tracing configuration");
+            }
+        }
+
+        return res;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean restoreDefaultConfiguration(@NotNull TracingConfigurationCoordinates coordinates) {
+        DistributedMetaStorage metaStore;
+
+        try {
+            metaStore = ctx.distributedMetastorage();
+        }
+        catch (Exception e) {
+            log.warning("Failed to restore tracing configuration for coordinates=[" + coordinates +
+                "] to default  — meta storage is not available.");
+
+            return false;
```

Comment:
I'd make all API methods void and wrap original exception into IgniteException and throw it out of the method.
I think it would be more transparent if user could observe the original reason why API call failed instead of searching it in the logs.

Reply:
It's not possible to make all of them void, cause 2 out of 4 return configuration, however void instead of boolean definitely has sense.
> wrap original exception into IgniteException and throw it out of the method.
Agreed.